### PR TITLE
Fix bug with recomputing quantities on incorrect grid

### PR DESCRIPTION
**Pull request by unalmis**
If you call `data = eq.compute(..., data=data)` the contract is that stuff already in `data` stays there. `master` violates this contract. At best it causes slow downs by recomputing things, at worst it computes quantities incorrectly.

### CODE REVIEW

**Review by dpanici — COMMENTED** [review]
Is there anyway to test this with existing code or is this too specific to the field line integral stuff?

**Comment by unalmis** [comment]
If you call `data = eq.compute(..., data=data)` the contract is that stuff already in `data` stays there. `master` violates this contract. At best it causes slow downs by recomputing things, at worst it computes quantities incorrectly. Yes this is needed for #1003. It's related to but distinct from #719 .

File: desc/equilibrium/equilibrium.py [hunk 1]
```diff
@@ -877,19 +877,31 @@ def compute(
...
             )
             # these should all be 0d quantities so don't need to compress/expand
             data0d = {key: val for key, val in data0d.items() if key in dep0d}
             data.update(data0d)
 
+        data0d_seed = (
+            {key: data[key] for key in data if data_index[p][key]["coordinates"] == ""}
+            if ((calc1dr or calc1dz) and override_grid)
+            else {}
+        )
```
Comment (f0uriest) [inline desc/equilibrium/equilibrium.py:904]: isn't this only needed above?

Comment (unalmis) [reply desc/equilibrium/equilibrium.py:904]: I think we should pass in seed data whenever we can (so if its already computed correctly and if it broadcasts with the calc1dx grids). For example, epsilon effective is function of rho and requires `R0` which is a scaler volume  quantity that would be computed wrong on a grid over a single surface.

**Comment by dpanici** [comment]
update master data

**Review by f0uriest — DISMISSED** [review]

**Review by f0uriest — COMMENTED** [review]
I think changing this line: 
https://github.com/PlasmaControl/DESC/blob/rc/grids/desc/equilibrium/equilibrium.py#L844
to 
```python
        deps = list(
            set(get_data_deps(names, obj=p, has_axis=grid.axis.size) + names)
            - data.keys()
        )
```

would also fix this issue. If the user wants to compute `x` which depends on `y` which in turn depends on `z`, and they pass in `y` then we shouldn't need to compute `z` at all.

**Comment by unalmis** [comment]
> I think changing this line: https://github.com/PlasmaControl/DESC/blob/rc/grids/desc/equilibrium/equilibrium.py#L844 to
> 
> ```python
>         deps = list(
>             set(get_data_deps(names, obj=p, has_axis=grid.axis.size) + names)
>             - data.keys()
>         )
> ```
> 
> would also fix this issue. If the user wants to compute `x` which depends on `y` which in turn depends on `z`, and they pass in `y` then we shouldn't need to compute `z` at all.

I added it, but subtracting out `data.keys()` as you suggest does:

If the user wants to compute `x` which depends on `y` ~~which in turn depends on `z`~~, and they pass in `y` then we shouldn't need to compute `y` at all.

The dependencies of `y` may still computed redundantly.

**Comment by f0uriest** [comment]
Ahhh ok yeah. In `desc.compute.utils.compute` we do things recursively, so in that case priming `data` with just the first order dependencies will avoid computing any extra stuff. But in `eq.compute` we first get the *full* list of dependencies (beyond just first order). That could eventually be fixed but doesn't have to be now.

**Review by f0uriest — APPROVED** [review]

**Pull request merged**